sdk/core/core-http/lib/util/utils.ts

```diff
@@ -241,3 +241,17 @@ export function replaceAll(
 export function isPrimitiveType(value: any): boolean {
   return (typeof value !== "object" && typeof value !== "function") || value === null;
 }
+
+/**
+ * @ignore
+ * Helper utility to clean up unintended characters that get appended by OS.
+ * @param str
+ */
+export function removeBOM(str: any) {
+  if (typeof str == "string") {
+    if (str.charCodeAt(0) === 0xfeff || str.charCodeAt(0) === 0xffef) {
```

Comment:
I believe that this checks for the BOM header when the encoding is expected to be UTF16. 
Would the correct check be for the following sequence of bytes `EF BB BF` ?

https://www.w3.org/International/questions/qa-byte-order-mark.en

Reply:
Looks like this is for utf-8, utf-16 seems to have some other considerations added in a different section

Reply:
Here is another resource -- there seems to be a difference between UTF8 and UTF16.

https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8
> The UTF-8 representation of the BOM is the (hexadecimal) byte sequence 0xEF,0xBB,0xBF

https://en.wikipedia.org/wiki/Byte_order_mark#UTF-16
> In UTF-16, a BOM (U+FEFF) may be placed as the first character of a file or character stream to indicate the endianness (byte order) of all the 16-bit code units of the file or stream. If an attempt is made to read this stream with the wrong endianness, the bytes will be swapped, thus delivering the character U+FFFE, which is defined by Unicode as a "non character" that should never appear in the text.

Can you share more information about the section you mention?

Reply:
In the earlier link at https://www.w3.org/International/questions/qa-byte-order-mark.en#bytheway
It says 
> If you use a UTF-16 encoding for your page (and we strongly recommend that you don't), there are some additional considerations.

So the directions mentioned in entirety of article seemed to refer to UTF-8

Reply:
In the same document, they have this:
> With a binary editor capable of displaying the hexadecimal byte values in the file, the UTF-8 signature displays as EF BB BF.

At this point, this is irrelevant as we don't have the BOM code anymore, but this is something to keep in mind for the SB changes.

Reply:
I created 3 files using notepad and saved them with different encodings (UTF8, UTF16 big endian, UTF16 little endian). 

Here is how they look like in a binary editor
UTF8
![image](https://user-images.githubusercontent.com/8396979/66960341-2103e080-f021-11e9-9137-76f0cdb5e0e8.png)
UTF16 Little Endian
![image](https://user-images.githubusercontent.com/8396979/66960380-2f51fc80-f021-11e9-9adb-918a079d9c6e.png)
UTF16 Big Endian
![image](https://user-images.githubusercontent.com/8396979/66960422-442e9000-f021-11e9-918c-cbe1dfbca00e.png)

Reply:
Since SB doesn't directly feed any files to input, it's not going to be necessary.
But when it comes to dealing with output sent by service, such as for `deserializationPolicy` and maybe others where the response from service is being parsed directly, would be good to have these in core-http I think.
Especially cause xml2js handles it in Node, but for in browser we are manually parsing and this isn't being handled implicitly.
So it seems general text stream handling things, wonder if Storage handles anything explicitly in the SDK.

Reply:
About the actual characters to use, above is interesting - unclear why the utf-16 code was being added in `azure-sb` - it might be possible that this specific character was noted in output. (Or maybe it was before xml2js (or the service itself) was updated to implicitly handle the BOM characters)